### DISCUSSION TITLE
Sharing singletons Between shell and MFEs

### DISCUSSION THREAD

**Discussion by RajathVenkatesh** [module-federation/module-federation-examples#2092] [Q&A]
I am using the Angular architects module federation and I would like to have singleton services created for communication between the shell and MFEs.

I know the approach in a monorepo project would be to use a Nx workspace shared library, but I'm really looking for a solution when there are multi repo projects.

Any advise/solution on how to deal with this?

**Comment by NikJ87** [comment]
Hi @RajathVenkatesh 

I'm also looking for similar kinda problem to use Singleton across MFEs from different Repos.

@ScriptedAlchemy  could you please hel us with it.

Thanks
Nik

**Reply by RajathVenkatesh** [reply]
@NikJ87 

Good day! I hope you doing good.

I think the only solution that I can think of as of now is to create an Angular library, let's call it a shared library.
(This has all the services, components etc you wish to share between apps) [The services need to be provideIn : 'platform' ]

Publish it to a private registry (preferred) or the public registry and consume it in your apps as a package (remotes as well as shell apps).

I still need to give this a try, will let you know how it goes once I do.

And also an important step is to share the package and use it as a singleton in both remote and shell webpack configs. Just like how you would the angular packages like router, core, http etc!
